src/abstractdatatable/join.jl

```diff
@@ -4,16 +4,13 @@
 
 # Like similar, but returns a array that can have nulls and is initialized with nulls
 similar_nullable{T}(dv::AbstractArray{T}, dims::Union{Int, Tuple{Vararg{Int}}}) =
-    (v = Vector{?T}(dims); fill!(v, null); return v)
+    (v = Vector{Union{T, Null}}(dims); fill!(v, null); return v)
```

Comment:
Shouldn't we use `similar(dv, Union{T, Null})`?

Reply:
I think that should work, I'll link to this in #66 and test it when I make updates there. Good suggestion!